client/build_mac_host.sh

```diff
@@ -1,25 +1,83 @@
 # Automates build process for Platform codegen on macOS.
 
-set -e
+set -ex
+
+# Defaults
+BUILD_TYPE="debug"
+
+# Parse CLI options
+while getopts ":ailrw" options; do
+    case $options in
+        a) FLAG_ANDROID=true ;;
+        i) FLAG_IOS=true ;;
+        l) FLAG_LINUX=true ;;
+        r) BUILD_TYPE="release" ;;
+        w) WORKAROUND_ROSETTA_ISSUE=true ;;
+    esac
+done
+
+## BUILD FOR CURRENT HOST ##
 
 CLIENT_DIR=$(realpath $(dirname $0))
 PLATFORM_DIR=$(realpath $CLIENT_DIR/..)
 RSHARP_DIR="$PLATFORM_DIR/crates/rsharp"
-
+HOST_TARGET_TRIPLE=$(rustc -vV | sed -n 's|host: ||p')
 
 # Build Platform core and demo libraries.
 cd $PLATFORM_DIR
-cargo build --all
+cross build --target $HOST_TARGET_TRIPLE -p unity_states
 
 # Generate C bindings.
 cargo test --all
 cargo run -p cs_codegen -- -f
 cargo run -p rsharp_codegen -- -f
 
-# Rename library from specific -> generic for symbol compatibility.
-# mv $PLATFORM_DIR/target/debug/libunity_states.dylib $PLATFORM_DIR/target/debug/libconstellation.dylib
+# Move library into symlinked location (if necessary)
+rsync $PLATFORM_DIR/target/$HOST_TARGET_TRIPLE/$BUILD_TYPE/libunity_states.dylib $PLATFORM_DIR/target/$HOST_TARGET_TRIPLE/libunity_states.dylib
+
+## CROSS-COMPILE TO LINUX
+
+if ! [ -z "$FLAG_LINUX" ]
+then
+    cross build --target x86_64-unknown-linux-gnu -p unity_states
+
+    # Move library into symlinked location (if needed)
+    rsync $PLATFORM_DIR/target/x86_64-unknown-linux-gnu/$BUILD_TYPE/libunity_states.so $PLATFORM_DIR/target/x86_64-unknown-linux-gnu/libunity_states.so
+fi
+
+## CROSS-COMPILE TO ANDROID
+
+if ! [ -z "$FLAG_ANDROID" ]
+then
+    cross build --target aarch64-linux-android -p unity_states
+    cross build --target armv7-linux-androideabi -p unity_states
+
+    # Move libraries into symlinked locations (if needed)
+    rsync $PLATFORM_DIR/target/aarch64-linux-android/$BUILD_TYPE/libunity_states.so $PLATFORM_DIR/target/aarch64-linux-android/libunity_states.so
+    rsync $PLATFORM_DIR/target/armv7-linux-androideabi/$BUILD_TYPE/libunity_states.so $PLATFORM_DIR/target/armv7-linux-androideabi/libunity_states.so
+fi
+
+## CROSS-COMPILE TO IOS
+
+if ! [ -z "$FLAG_IOS" ]
+then
+    cross build --target aarch64-apple-ios -p unity_states
+
+    # Move library into symlinked location (if needed)
+    rsync $PLATFORM_DIR/target/aarch64-apple-ios/$BUILD_TYPE/libunity_states.a $PLATFORM_DIR/target/aarch64-apple-ios/libunity_states.a
+fi
+
+## BINDINGS FOR DYNAMICALLY-LINKED TARGETS ##
+
+if ! [ -z "$WORKAROUND_ROSETTA_ISSUE" ]
```

Comment:
This works around an issue I've been having with Rosetta 2 on my Mac. Based on the number of forum posts it is relatively uncommon, but it renders Rosetta-powered processes useless once the error occurs (which is not every time). I've found that the errors occur when Rosetta caches an old version of the `.aot` file and attempts to run that after the original code has changed. In our project, the `.aot` files Rosetta compiles from our code are based on the `.csproj` filename. Short of a clean reinstall of macOS or restarting my machine constantly, the only reliable way I've found around these errors is to rename the process on each iteration. This is a temporary fix but I needed something to avoid an OS re-install while in SF :P

Reply:
> clean reinstall of macOS

![](https://c.tenor.com/xMc6o2-3-xkAAAAC/excuse-me-cry.gif)

Reply:
You should add a comment that this avoids needing to reinstall macos, and a link to a relevant github issue (or even this comment)